README.md

```diff
@@ -134,6 +134,49 @@ becomes
       extra-lib-dirs:
         lib/darwin
 
+Multiple conditionals must fall under one `when`, for example:
+
+    when:
+      - condition: os(darwin)
+        extra-lib-dirs: lib/darwin
+      - condition: os(linux)
+        extra-lib-dirs: /usr/lib/x64
+
+becomes
+
+    if os(darwin)
+      extra-lib-dirs:
+        lib/darwin
+    if os(linux)
+      extra-lib-dirs:
+        /usr/lib/x64
+
+If conditionals are declared under different `when` clauses only the last one
+will make it to the `.cabal` file:
+
+    when:
+      - condition: os(darwin)
+        extra-lib-dirs: lib/darwin
+    when:
+      - condition: os(linux)
+        extra-lib-dirs: /usr/lib/x64
+
+becomes:
+
+    if os(linux)
+      extra-lib-dirs:
+        /usr/lib/x64
+
+Unlike with `.cabal` files compound predicates need to be in parens, for example:
+
+    when:
+      - condition: (!os(darwin) && !os(windows))
```

Comment:
I believe the issue here is not that you have "compound predicate" here.  It's merely the `!`, which in YAML must be quoted when at the beginning of a string, say:

    condition: "!os(darwin) && !os(windows)"